modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheConfigurationEnricher.java

```diff
@@ -141,7 +142,15 @@ public CacheGroupDescriptor enrich(CacheGroupDescriptor desc, boolean affinityNo
             }
         }
         catch (Exception e) {
-            throw new IgniteException("Failed to enrich cache configuration [cacheName=" + ccfg.getName() + "]", e);
+            StringBuilder msgBldr = new StringBuilder("Failed to enrich cache configuration [cacheName=")
```

Comment:
I think that we need to update `CacheConfigurationSplitterImpl` as well. Let's assume the following RU scenario:
 - start one server node using gg version that does not support serializing `sqlFunCls`
 - start a new node with the fix
 - request a new dynamic cache from the new node
 
 In my understanding, it can be a problem for the first node to properly process the request because the real value of `sqlFuncCls` will be placed in the enrichment.

Reply:
@sk0x50 I wrote exactly that compatibility test and it seems working well. The reason for that is in `CacheConfigurationEnricher` code: it doesn't care about which fields to process and which not and deserialize them all and set into cache configuration.

So no additional logic is needed here.